compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,15 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
```

Comment:
How should this behave for `union`s (and other items)? I think they should behave like `struct`s, so this should be correct, but I'm not sure.

Reply:
Yes for fields in unions and structs the current behavior should be correct (and we don't have other fields besides them and fields in variants).